src/skmultiflow/data/base_generator.py

```diff
@@ -2,26 +2,26 @@
 from skmultiflow.core.base_object import BaseObject
 
 
-class Stream(BaseObject, metaclass=ABCMeta):
-    """ The abstract class setting up the minimum requirements of a stream,
+class BaseGenerator(BaseObject, metaclass=ABCMeta):
```

Comment:
This renaming could raise some confusion given that not all extended-classes are generators. Suggestions: `SMKStream`, `BaseStream`, `StreamMixin`,...

This is not limited to available methods, but also considering future sources such as kafka, tcp, http, etc.

Reply:
@jacobmontiel do you have an example ? This was to avoid confusion with streamz.Stream. 
IMO the extended-classes I have seen literally are generators, but I may be confused ...

Reply:
Currently, there are two types of data sources supported: generators and data batches (files and raw data).

In this case, FileStream and DataStream are not generators. Future sources (via streamz) that come to mind include Kafka, TCP which also do not fall in the generator category.

During the first stage of development, we focused on generators because they are cheap (memory-wise). However, the next step is to allow multiple data sources. Generators are nice for reproducibility, but in order to make `skmultiflow` useful for real-world applications, we need to provide the user with more options.